Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -510,8 +589,51 @@ private void RemovePointer(IMixedRealityPointer pointer)
             pointers.Remove(pointer);
         }
 
+        /// <summary>
+        /// Access the PointerManager with all the interactive pointer data.
+        /// </summary>
+        /// <returns></returns>
+        public InteractablePointerDataManager GetPointerManager()
+        {
+            return pointerDataManager;
+        }
+
         #endregion PointerManagement
 
+        #region SourceManagement
+
+        /// <summary>
+        /// Adds a source to sources
+        /// </summary>
+        /// <param name="source"></param>
+        private void AddSource(SourceStateEventData source)
+        {
+            if (!sources.Contains(source))
```

Comment:
Why not just use a hashset, if you want to have a list of unique sources? A hashset will allow you to add and check for membership in O(1) time, where as this approach is O(N). Note it doesn't really matter for a list this small, but it will also simplify your code (you won't need these two methos, just you just do sources.Add() and sources.Remove() directly.

Reply:
using hashSet now.